File: Algorithms/Mathematical Algorithms/factorial.rb

# Ruby function to get the factorial of an integer
def factorial(number)
  return 1 if number <= 1
  number * factorial(number - 1)
end

# How to run:
# $ ruby factorial.rb
puts "Factorial of an Integer"
puts "Enter an integer: "
number = gets.chomp
puts "The factorial of the Integer #{number} is: #{factorial(number.to_i)}"

File: factorial.c
#include<stdio.h>
double fact(int n);
void main()
{
	int n;
	printf("Enter number");
	scanf("%d",&n);
	printf("Factorial= %.2li",fact(n));
	
}
double fact(int n)
{
	int f=1,i;
	for(i=1;i<=n;i++)
	{
		f=f*i;
	}
	return f;
}
